### PR TITLE
feat(github): Show error message when github fork fails

### PR DESCRIPTION
**Pull request by jamespohalloran**
Show error message when GitHub fork fails. This might occur is the repo url is incorrect or private.
Screen Shot 2020-04-17 at 10 53 00 AM

### CODE REVIEW

File: packages/react-tinacms-github/src/components/AsyncButton.tsx [hunk 1]
```diff
@@ -31,8 +31,12 @@ export const AsyncButton = ({ name, primary, action }: ButtonProps) => {
 
   const onClick = useCallback(async () => {
     setSubmitting(true)
-    await action()
-    setSubmitting(false)
+    try {
+      await action()
+    } catch (e) {
+      setSubmitting(false)
+      throw e
```
Comment (ncphillips) [inline packages/react-tinacms-github/src/components/AsyncButton.tsx:38]:
```suggestion
      await action()
      setSubmitting(false)
    } catch (e) {
      setSubmitting(false)
      throw e
```

File: packages/react-tinacms-github/src/github-editing-context/GithubAuthModal.tsx [hunk 2]
```diff
@@ -71,7 +81,15 @@ const GithubAuthModal = ({
...
+            try {
+              full_name = (await cms.api.github.createFork()).full_name
+            } catch (e) {
+              setError(
+                'Forking repository failed. Are you sure you have access?'
+              )
+              throw e
+            }
             setForkName(full_name)
             onUpdateAuthState()
```
Comment (ncphillips) [inline packages/react-tinacms-github/src/github-editing-context/GithubAuthModal.tsx:94]:
```suggestion
            try {
              const full_name = (await cms.api.github.createFork()).full_name
              setForkName(full_name)
              onUpdateAuthState()
            } catch (e) {
              setError(
                'Forking repository failed. Are you sure you have access?'
              )
              throw e
            }

```

Comment (ncphillips) [reply packages/react-tinacms-github/src/github-editing-context/GithubAuthModal.tsx:94]: This diff sucks. It just moves everything inside the `try`

File: packages/react-tinacms-github/src/components/AsyncButton.tsx (hunk 1 above)
Comment (ncphillips) [reply packages/react-tinacms-github/src/components/AsyncButton.tsx:38]: setSubmitting false if action is successful

**Comment by DirtyF** [comment]
> This might occur is the repo url is incorrect or private.

**Pull request merged**